src/command_modules/azure-cli-network/azure/cli/command_modules/network/_format.py

```diff
@@ -303,3 +303,88 @@ def transform_vnet_gateway_bgp_peer_table(result):
             ('MessagesReceived', item['messages_received'] or '')
         ]))
     return transformed
+
+
+def transform_privatedns_zone_table_output(result):
+    is_list = isinstance(result, list)
+
+    if not is_list:
+        result = [result]
+
+    final_result = []
+    for item in result:
+        new_item = OrderedDict([
+            ('ZoneName', item['name']),
+            ('ResourceGroup', item['resourceGroup']),
+            ('RecordSets', item['numberOfRecordSets']),
+            ('MaxRecordSets', item['maxNumberOfRecordSets']),
+            ('VirtualNetworkLinks', item['numberOfVirtualNetworkLinks']),
+            ('MaxVirtualNetworkLinks', item['maxNumberOfVirtualNetworkLinks']),
+            ('VirtualNetworkLinksWithRegistration',
+             item['numberOfVirtualNetworkLinksWithRegistration']),
+            ('MaxVirtualNetworkLinksWithRegistration',
+             item['maxNumberOfVirtualNetworkLinksWithRegistration']),
+            ('ProvisioningState', item['provisioningState'])
+        ])
+        final_result.append(new_item)
+
+    return final_result if is_list else final_result[0]
+
+
+def transform_privatedns_link_table_output(result):
+    is_list = isinstance(result, list)
+
+    if not is_list:
+        result = [result]
+
+    final_result = []
+    for item in result:
+        new_item = OrderedDict([
+            ('LinkName', item['name']),
+            ('ResourceGroup', item['resourceGroup']),
+            ('RegistrationEnabled', item['registrationEnabled']),
+            ('VirtualNetwork', item['virtualNetwork']['id']),
+            ('LinkState', item['virtualNetworkLinkState']),
+            ('ProvisioningState', item['provisioningState'])
+        ])
+        final_result.append(new_item)
+
+    return final_result if is_list else final_result[0]
+
+
+def transform_privatedns_record_set_output(result):
+    from azure.mgmt.privatedns.models import RecordSetPaged
+
+    def _strip_null_records(item):
+        for prop in [x for x in dir(item) if 'record' in x]:
+            if not getattr(item, prop):
+                delattr(item, prop)
+
+    if isinstance(result, RecordSetPaged):
+        result = list(result)
+        for item in result:
+            _strip_null_records(item)
+    else:
+        _strip_null_records(result)
+
+    return result
+
+
+def transform_privatedns_record_set_table_output(result):
```

Comment:
`transform_privatedns_record_set_table_output` is separate because the PrivateDns `RecordSet` model is different from `RecordSet` model of Dns. And these models can change in the future and more properties can be added. Sharing the function among the two only will make it complicated as we would have to use flags or something to denote which model it needs to transform which is not a clean solution. It is better to keep it separate as it is a completely new and separate service.